File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumDifference = function(nums) {
    
    let left = 0, right = 1, maxDiff = -1
    
    while(right < nums.length){
        if(nums[left] >= nums[right]){
            left = right
            right++
        }else{
            let tempDiff = nums[right] - nums[left]
            maxDiff = Math.max(tempDiff, maxDiff)
            right++
        }
    }
    
    return maxDiff
    
};

File: 2016-maximum-difference-between-increasing-elements/2016-maximum-difference-between-increasing-elements.java
class Solution {
    public int maximumDifference(int[] nums) {
        
        int left = 0, right = 1, maxDiff = -1;
        while(right < nums.length){
            if(nums[left] >= nums[right]){
                left = right;
                right++;
            }else{
                int tempDiff = nums[right] - nums[left];
                maxDiff = Math.max(maxDiff, tempDiff);
                right++;
            }
        }
        return maxDiff;
    }
}